### PR TITLE
Fix bug with modification of `AVAILABLE_COLORMAPS` when iterate over its. 

### PR DESCRIPTION
**Pull request by Czaki**
# Description

This is my proposition to fix bug from #2182., 
This PR introduces a global lock and protect the content of `ensure_colormap` function to prevent changes in `AVAILABLE_COLORMAPS` dict when iterating over its content. 

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# References

https://docs.python.org/3/library/threading.html#using-locks-conditions-and-semaphores-in-the-with-statement

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by sofroniewn — COMMENTED** [review]
@Czaki the diff for this PR is a little hard to parse. Is the only change that the inside of this function is now fully located inside the `with AVAILABLE_COLORMAPS_LOCK:` or did the inside bits of that function change too?

**Comment by Czaki** [comment]
@sofroniewn only adding `with` clause and indent rest of function code (except return clause).

**Review by sofroniewn — APPROVED** [review]
This looks good to me! The lock makes sense, thanks for adding the link to the python docs.

Curious if @tlambert03 @jni have any additional thoughts before merge?

**Review by tlambert03 — APPROVED** [review]
